packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -268,9 +268,19 @@ export class AssetStaging extends Construct {
    */
   private stageByCopying(): StagedAsset {
     const assetHash = this.calculateHash(this.hashType);
-    const stagedPath = this.stagingDisabled
-      ? this.sourcePath
-      : path.resolve(this.assetOutdir, renderAssetFilename(assetHash, path.extname(this.sourcePath)));
+    let stagedPath: string;
+
+    if (this.stagingDisabled) {
+      stagedPath = this.sourcePath;
+    } else {
+      let extensionName: string = path.extname(this.sourcePath);
+      const sourceName: string = path.basename(this.sourcePath).replace(extensionName, '');
+      const doubleArchive = ARCHIVE_EXTENSIONS.includes(path.extname(sourceName)) ? true : false;
```

Comment:
`Array.prototype.includes()` already returns `true` or `false`. This ternary construction is unnecessary.